### PR TITLE
add templates

### PR DESCRIPTION
**Pull request by mjiang102628**
- Split
- HTLC
- DynamicFee

### CODE REVIEW

File: algosdk/template.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+        amt_1 = 0
+        amt_2 = 0
+        
+        gcd = math.gcd(self.ratn, self.ratd)
+        ratn = self.ratn // gcd
+        ratd = self.ratd // gcd
+
+        if amount % ratd == 0:
+            amt_1 = amount // ratd * ratn
+            amt_2 = amount - amt_2
```
Comment (EvanJRichard) [inline algosdk/template.py]: I know this is a draft but I spotted this while looking at how you were implementing `precise` logic - `amt_2 = amount - amt_1`

Comment (EvanJRichard) [reply algosdk/template.py]: Also I like how you're implementing `precise`, copying that for js-sdk

File: algosdk/template.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    a split where the first address receives 30 percent and the second receives 70, 
+    set ratn and ratd to 30 and 100, respectively. 
+    Split also have an expiry round, in which the owner can transfer back the assets.
+    ...
+
+    Arguments:
+        owner (str): an address that can receive the asset after the expiry round
+        receiver_1 (str): first address to receive assets 
+        receiver_2 (str): second address to receive assets 
+        ratn (int): the numerator of the first address fraction
```
Comment (algonautshant) [inline algosdk/template.py]: The comments are updated in: https://github.com/algorand/go-algorand/pull/536/files
May need to be updated here. 
Also, is this just for assets?

File: algosdk/template.py [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+            res = replace(res, val, offsets[i], 1)
+
+        elif val_type == "address":
+            val = encoding.decode_address(val)
+            res = replace(res, val, offsets[i], 32)
+
+        elif val_type == "base64":
+            val = bytes(base64.b64decode(val))
+            buf = []
+            dec_len = put_uvarint(buf, len(val)) + len(val)
```
Comment (algonautshant) [inline algosdk/template.py]: dec_len = put_uvarint(buf, len(val)) + len(val) **-1** ?
Since the base64 placeholder is 2, and 1 is subtracted in line 432.

**Comment by mjiang102628** [comment]
closing; has been split into three PRs

**Pull request closed without merging**